tests/integration/zenodo_test.py

```diff
@@ -0,0 +1,212 @@
+        },
+        "created_file.txt": {
+            "updated": b"This file should created during deposition update.",
+        },
+    }
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_dir_path = Path(tmp_dir)
+
+        file_paths = {
+            "original": tmp_dir_path / "original",
+            "updated": tmp_dir_path / "updated",
+        }
+
+        # Create directories
+        file_paths["original"].mkdir()
+        file_paths["updated"].mkdir()
+
+        # Loop through files and create them on local disk
+        files = {"original": [], "updated": []}
+        for filename, data in file_data.items():
+            for dep_type, contents in data.items():
+                path = file_paths[dep_type] / filename
+                # Add to files
+                files[dep_type].append({"path": path, "contents": contents})
+
+                with open(path, "wb") as f:
+                    f.write(contents)
+
+        yield files
+
+
+@pytest.fixture(name="test_settings")
+def create_test_settings_file():
+    """Create temporary DOI settings file."""
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        settings_file = Path(tmp_dir) / "settings.yaml"
+        with open(settings_file, "w") as f:
+            f.writelines(["fake_dataset:\n", "    sandbox_doi: null"])
+
+        yield settings_file
+
+
+@pytest.mark.asyncio
+async def test_zenodo_workflow(
+    session: aiohttp.ClientSession,
+    upload_key: str,
+    publish_key: str,
+    test_settings: Path,
+    test_files: dict[str, list[dict[str, str]]],
+    deposition_metadata: DepositionMetadata,
+    datapackage: DataPackage,
+    mocker,
+):
+    """Test the entire zenodo client workflow."""
+    from pudl_archiver.zenodo.api_client import ZenodoClient
+
+    client = ZenodoClient(test_settings, session, upload_key, publish_key, testing=True)
+
+    # Mock out creating deposition metadata with fake data source
+    deposition_metadata_mock = mocker.MagicMock(return_value=deposition_metadata)
+    mocker.patch(
+        "pudl_archiver.zenodo.api_client.DepositionMetadata.from_data_source",
+        new=deposition_metadata_mock,
+    )
+
+    # Mock out creating datapackage with fake data source
+    datapackage_mock = mocker.MagicMock(return_value=datapackage)
+    mocker.patch(
+        "pudl_archiver.zenodo.api_client.DataPackage.from_filelist",
+        new=datapackage_mock,
+    )
+
+    # Create new deposition and add files
+    async with client.deposition_interface(
+        "pudl_test", initialize=True, dry_run=False
+    ) as interface:
+        resources = {
+            file_data["path"].name: ResourceInfo(
+                local_path=file_data["path"], partitions={}
+            )
+            for file_data in test_files["original"]
+        }
+        await interface.add_files(resources)
+
+    # Wait before trying to access newly created deposition
+    time.sleep(1.0)
```

Comment:
If we add smart retrying logic in the future we can probably avoid these `time.sleep`s.